### PR TITLE
Add github action to run flake8

### PR DESCRIPTION
**Pull request by ManuelStardust**
## 🛠️ Changes
- Add github flow to flake8 check
- Add github page link in README

## 📝 Associated issues
#6

### CODE REVIEW

File: .github/workflows/flake8.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: Flake8 Check
+
+on:
+  pull_request:
+    branches:
+      - master
```
Comment (erikasv) [inline .github/workflows/flake8.yml]: this branch doesn't exists, actually I think this PR should run the action

File: .github/workflows/flake8.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+jobs:
+  flake8:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
```
Comment (erikasv) [inline .github/workflows/flake8.yml]: this version is very old, I think they are in 4 now

File: .github/workflows/flake8.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
```
Comment (erikasv) [inline .github/workflows/flake8.yml]: This version is also very old

File: .github/workflows/flake8.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        with:
+          python-version: 3.x
```
Comment (erikasv) [inline .github/workflows/flake8.yml]: I don't think is a good idea to let that version so wide, check if the action can take the version from pyproject.tml, so we don't have to write that number in various places.

And this reminded me that we chose a target version kind of old (https://github.com/PEM-Humboldt/stac-data-tools/blob/main/pyproject.toml#L3). Change that and this one to a newer one (3.11 or 3.12) (https://devguide.python.org/versions/)

File: .github/workflows/flake8.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+      - name: Install dependencies
+        run: pip install flake8
```
Comment (erikasv) [inline .github/workflows/flake8.yml]: There is an action for flake8, wouldn't be better to use it? (https://github.com/py-actions/flake8) or why not?

**Review by erikasv — CHANGES_REQUESTED** [review]

File: .github/workflows/flake8.yml (hunk 1 above)
Comment (ManuelStardust) [reply .github/workflows/flake8.yml]: Done

File: .github/workflows/flake8.yml (hunk 2 above)
Comment (ManuelStardust) [reply .github/workflows/flake8.yml]: Done

File: .github/workflows/flake8.yml (hunk 3 above)
Comment (ManuelStardust) [reply .github/workflows/flake8.yml]: Done

File: .github/workflows/flake8.yml (hunk 4 above)
Comment (ManuelStardust) [reply .github/workflows/flake8.yml]: Done

File: .github/workflows/flake8.yml (hunk 5 above)
Comment (ManuelStardust) [reply .github/workflows/flake8.yml]: Thank you, I had not seen that workflow, I already updated the one I had created based on that

**Review by erikasv — COMMENTED** [review]
I see everything ok now, but there is a warning in the workflow execution `Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: actions/setup-python@v4. ...` could you check if there is a newer version of that action?

**Comment by ManuelStardust** [comment]
> I see everything ok now, but there is a warning in the workflow execution `Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: actions/setup-python@v4. ...` could you check if there is a newer version of that action?

Yes, there is v5. Solved.

**Review by erikasv — APPROVED** [review]

**Pull request merged**